packages/flutter_ai_providers/lib/src/providers/openwebui.dart

```diff
@@ -0,0 +1,307 @@
+    final llmMessage = messages.last;
+    if(files.isNotEmpty) {
+      for (var file in files) {
+        await _handleAttachment(file);
+      }
+    }
+
+    final httpRequest = http.Request('POST', Uri.parse("$_host/chat/completions"))
+      ..headers.addAll({
+        if(_apiKey != null) 'Authorization': 'Bearer $_apiKey',
+        'Content-Type': 'application/json',
+      })
+      ..body = _OwuiChatRequest(
+        model: _model,
+        messages: messages.where((m) => m.text != null).toList(),
+        files: _fileAttachments,
+        images: _imageAttachments,
+      ).toJsonString();
+
+    final httpResponse = await http.Client().send(httpRequest);
+
+    if (httpResponse.statusCode == 200) {
+      await for (var text in httpResponse.stream.transform(utf8.decoder)) {
+        final messages = text.split('\n');
+        for (var message in messages) {
+          if (message.startsWith('data: [DONE]')) {
+            return;
+          }
+          if (message.isEmpty) continue;
+          final cleanedMessage = message.replaceFirst('data: ', '').trim();
+          try {
+            final chatResponse = _OwuiChatResponse.fromJsonString(cleanedMessage);
+            for (var choice in chatResponse.choices) {
+              llmMessage.append(choice.message.text ?? '');
+              yield choice.message.text ?? '';
+            }
+          } catch (e) {
+            // just skip?
+          }
+        }
+        llmMessage.append('');
+        yield '';
+      }
+    } else {
+      throw Exception('HTTP request failed. Status: ${httpResponse.statusCode}, Reason: ${httpResponse.reasonPhrase}');
+    }
+  }
+
+  Future<void> _handleAttachment(Attachment attachment) async {
+    if(attachment is ImageFileAttachment) {
+      _imageAttachments.add(_OwuiImageAttachment.fromImageAttachment(attachment));
+      log(_imageAttachments.toString());
+    } else if(attachment is FileAttachment) {
+      final uri = Uri.parse('$_host/v1/files/'); // Replace with your OpenWebUI endpoint
+      final request = http.MultipartRequest('POST', uri)
+        ..headers.addAll({
+          if(_apiKey != null) 'Authorization': 'Bearer $_apiKey',
+          'Content-Type': 'multipart/form-data',
+          'Accept': 'application/json',
+        })
+        ..files.add(http.MultipartFile.fromBytes('file', attachment.bytes, filename: attachment.name));
+
+      final response = await request.send();
+      if (response.statusCode == 200) {
+        final responseBody = await response.stream.bytesToString();
+        final jsonResponse = json.decode(responseBody);
+        _fileAttachments.add(_OwuiFileAttachment(
+          type: 'file',
+          id: jsonResponse['id'].toString(),
+        ));
+      } else {
+        throw Exception('Failed to upload file: ${response.reasonPhrase}');
+      }
+    }
+  }
+
+  @override
+  get history => List.from(_history);
```

Comment:
Let's specify the type, as it's a public API (`Iterable<ChatMessage> get history...`)

Reply:
Done